source/src/enet_encap/encap.c

```diff
@@ -523,7 +532,11 @@ void DetermineDelayTime(const EipByte *const buffer_start,
   } else if (kListIdentityMinimumDelayTime > maximum_delay_time) {       /* if maximum_delay_time is between 1 and 500ms set it to 500ms */
     maximum_delay_time = kListIdentityMinimumDelayTime;
   }
-  delayed_message_buffer->time_out = (maximum_delay_time * rand() ) / RAND_MAX;      /* Sets delay time between 0 and maximum_delay_time */
+  /* Sets delay time between 0 and 90% of maximum_delay_time */
+  /* 90% is taken to have some room for response to reach the requester 
+   * within maximum_delay_time sent by requester
+   */
+  delayed_message_buffer->time_out = rand() % ((maximum_delay_time * kListIdentityDelayXmitBuffer)/100);

```

Comment:
I am not sure about this factor. I think the scanner should cope with the possible additional delay from the response, and not the adapter. Also, although I am not completely sure if this is the problem, I sometimes get an additional timeout error for class 3 connections in the CT tool, which I never had before.